src/com/amazon/ion/impl/IonReaderContinuableApplicationBinary.java

```diff
@@ -181,6 +197,218 @@ private SymbolTable getSystemSymbolTable() {
         return SharedSymbolTable.getSystemSymbolTable(getIonMajorVersion());
     }
 
+    /**
+     * Read-only snapshot of the local symbol table at the reader's current position.
+     */
+    private class LocalSymbolTableSnapshot implements SymbolTable, SymbolTableAsStruct {
+
+        // The system symbol table.
+        private final SymbolTable system = IonReaderContinuableApplicationBinary.this.getSystemSymbolTable();
+
+        // The max ID of this local symbol table.
+        private final int maxId;
+
+        // The shared symbol tables imported by this local symbol table.
+        private final LocalSymbolTableImports importedTables;
+
+        // Map representation of this symbol table. Keys are symbol text; values are the lowest symbol ID that maps
+        // to that text.
+        final Map<String, Integer> mapView;
+
+        // List representation of this symbol table, indexed by symbol ID.
+        final String[] listView;
+
+        private SymbolTableStructCache structCache = null;
+
+        LocalSymbolTableSnapshot() {
+            int importsMaxId = imports.getMaxId();
+            int numberOfLocalSymbols = localSymbolMaxId + 1;
+            // Note: 'imports' is immutable, so a clone is not needed.
+            importedTables = imports;
+            maxId = importsMaxId + numberOfLocalSymbols;
+            // Map with initial size the number of symbols and load factor 1, meaning it must be full before growing.
+            // It is not expected to grow.
```

Comment:
I had updated the code but not the comment. Load factor 1 with the map completely full results in worse lookup performance than loading the HashMap to 75% capacity and ensuring no growth. I moved the comment so that it sits on top of the HashMap and revised it to the following:

```
// Map with initial size and load factor set so that it will not grow unconditionally when it is filled.
// Note: using the default load factor of 0.75 results in better lookup performance than using 1.0 and
// filling the map to capacity.
```